### DISCUSSION TITLE
Loop to repeat the all timeline

### DISCUSSION THREAD

**Discussion by valentinavellani** [jspsych/jsPsych#1098] [Q&A] [answered]
Hi All,
I created a timeline and now I need to repeat it a given n of trials. 
This is the lines I inserted: I created a loop variable which repeats the whole timeline if n_trials is lower than 3. Then I appended this trial to the main timeline so it runs at the end of it.
```
// repeate the all timeline if repeat loop is true
var n_trial=0
var loop_node1 = {
    timeline: [timeline],
    loop_function: function(){
         n_trial=n_trial+1 //INCREASE N OF TRIAL BY ONE
        if(n_trial<3) {
            return true;
        } else {
            return false;
        }
    }
}

timeline.push(loop_node1)
```

However this does not work and I don't get why.

Thanks for your help!
Valentina

**Comment by becky-gilbert** [comment, marked as answer]
Hi @valentinavellani, your loop_function looks like it should work.

One thing to note is that the loop node's timeline should be an array of _objects_, and not an array containing another array. So if your `timeline` variable is already an array, then you can use it as the loop node's timeline without the square brackets.

Another thing I noticed that that you're pushing `loop_node1` to the same `timeline` array that you're using in the loop node. The timeline that is passed to `jsPsych.init` should be a separate array from the one used in your loop node. So altogether it might look like this:
```javascript
var timeline = []; // main experiment timeline

var loop_timeline = [trial1, trial2, trial3]; // trial objects that will be looped over
var n_trial=0
var loop_node1 = {
    timeline: loop_timeline,
    loop_function: function(){
         n_trial=n_trial+1 //INCREASE N OF TRIAL BY ONE
        if(n_trial<3) {
            return true;
        } else {
            return false;
        }
    }
}
timeline.push(loop_node1);

jsPsych.init({
  timeline: timeline
});
```

**Reply by valentinavellani** [reply]
This has worked perfectly!

Many thanks Becky! :)
